app/workers/late_reservation_checker.rb

```diff
@@ -0,0 +1,22 @@
+class LateReservationChecker
+  include Sidekiq::Worker
+  include Sidetiq::Schedulable
+
+  recurrence { daily }
+
+  def perform
+    late_reservations = Reservation.reserved.joins(gift_request: {list: :event}).where("events.end_date < ? OR delinquent IS TRUE", Date.current + 7.days)
+    delinquent_donors = Hash.new{0}
+    #for each loop, going through all reservations looking for common donors
+    late_reservations.each do |reservation|
+      delinquent_donors[reservation.donor] += 1
+
+    end
+    delinquent_donors.each do |donor, delinquencies|
+      #insert how many gifts they forgot to send
+      #If you have time, add recipient and what the gift was
+      LateReservationMailer.forgot_to_buy_gift_email(donor.user).deliver_now
```

Comment:
pass the number of delinquencies to the mailer and include it in the body e.g. "you have 3 reserved gifts you have not marked as shipped"